src/main/test/com/epam/esm/service/GiftCertificateServiceTest.java

```diff
@@ -0,0 +1,109 @@
+import java.util.List;
+import java.util.Optional;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Mockito.*;
+
+public class GiftCertificateServiceTest {
+
+    private static GiftCertificateDao giftCertificateDao;
+    private static GiftCertificateTagDao giftCertificateTagDao;
+    private static TagDao tagDao;
+    private static GiftCertificateValidator validator;
+    private static GiftCertificateDtoMapper dtoMapper;
+    private static GiftCertificate giftCertificate;
+    private static GiftCertificateDto giftCertificateDto;
+    private static GiftCertificateService service;
+    private static Optional<GiftCertificate> optionalGiftCertificate;
+    private static List<GiftCertificate> giftCertificates;
+    private static List<GiftCertificateDto> giftCertificateDtos;
+    private static Environment environment;
+
+    @BeforeClass
+    public static void init() {
+        giftCertificateDao = Mockito.mock(GiftCertificateDao.class);
+        giftCertificateTagDao = Mockito.mock(GiftCertificateTagDao.class);
+        tagDao = Mockito.mock(TagDao.class);
+        validator = Mockito.mock(GiftCertificateValidator.class);
+        dtoMapper = Mockito.mock(GiftCertificateDtoMapper.class);
+        giftCertificate = Mockito.mock(GiftCertificate.class);
+        giftCertificateDto = Mockito.mock(GiftCertificateDto.class);
+        environment = Mockito.mock(Environment.class);
+        service = new GiftCertificateService(giftCertificateDao,giftCertificateTagDao, tagDao, validator, dtoMapper, environment);
+        optionalGiftCertificate = Optional.of(giftCertificate);
+        giftCertificates = new ArrayList<>();
+        giftCertificateDtos = new ArrayList<>();
+        when(giftCertificateDao.findById(anyLong())).thenReturn(optionalGiftCertificate);
+        when(dtoMapper.map(any())).thenReturn(giftCertificateDto);
+        when(dtoMapper.unmap(any())).thenReturn(giftCertificate);
+        when(validator.validateCreating(any())).thenReturn(true);
+        when((validator.validateUpdating(any()))).thenReturn(true);
+        when(giftCertificateDao.getGiftCertificates(anyString(), anyString(), anyString(), anyString(), anyString())).thenReturn(giftCertificates);
+    }
+
+    @Test
+    public void testShouldReturnGiftCertificate() throws EntityNotExistsException {
+        //when
+        GiftCertificateDto giftCertificateDto = service.getGiftCertificate(1L);
+
+        //then
+        Assert.assertNotNull(giftCertificateDto);
+    }
+
+
+    @Test
+    public void testShouldDeleteEntity() throws EntityNotExistsException {
+        //when
+        service.deleteEntity(1L);
+
+        //then
+        verify(giftCertificateTagDao, times(1)).deleteGiftCertificateById(anyLong());
+        verify(giftCertificateDao, times(1)).deleteById(anyLong());
+    }
+
+    @Test
+    public void testShouldCreateGiftCertificate() throws EntityNotExistsException, BadEntityException {
```

Comment:
You should probably add more checks to some test methods. Like in this one and `testShouldUpdateGiftCertificate` you can compare some fields of the returned object and dto to see that entity objects were populated/created/updated correctly.

Reply:
Added checks in test classes in order to ensure that test works properly